iks-life-preserver/main.tf

```diff
@@ -0,0 +1,36 @@
+provider "ibm" {
+  ibmcloud_api_key = var.apikey
+}
+
+resource "ibm_iam_service_id" "lifeperserver" {
+  name = "life-preserver-id-${var.reference}"
+}
+
+resource "ibm_iam_service_policy" "life-preserver-policy" {
+  depends_on = [ ibm_iam_service_id.lifeperserver ]
+  iam_service_id = ibm_iam_service_id.lifeperserver.id
+  roles        = [ "Viewer", "Reader"]
+
+  resources {
+    service              = "containers-kubernetes"
+    resource_instance_id = var.cluster
+  }
+}
+
+data "external" "apikey" {
+  depends_on = [ ibm_iam_service_id.lifeperserver ]
+  program = ["bash", "${path.root}/createKey.sh", var.apikey, var.account, var.reference]
+}
+
+
+# resource "ibm_iam_user_policy" "policy" {
```

Comment:
If you're going to keep the commented-out section, I'd suggest providing some in-line comment explaining when it might be used. If it isn't needed, then removed.

Reply:
Done in https://github.com/IBM-Cloud/kube-samples/pull/94/commits/85c699c6b8bdc2d945b7658b2c63604c40e5fa0d